### PR TITLE
Setup postgres repo

### PR DESCRIPTION
**Pull request by conquext**
The PR contains basic setup for the postgres repo extending from the abstract repo class.

### CODE REVIEW

File: chaindexing-postgres/src/drizzle/db.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Pool } from 'pg';
+
+// TODO: Get connection string from env
+export const pool = new Pool({
+  connectionString: 'postgres://user:password@host:port/db'
```
Comment (Jurshsmith) [inline chaindexing-postgres/src/drizzle/db.ts:5]: A bit early, but let's set the maximum parallel connections allowed:

```suggestion
  connectionString: 'postgres://user:password@host:port/db',
  max: 2
```

File: chaindexing-postgres/src/drizzle/schema.ts [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+    startBlockNumber: integer('start_block_number'),
+    address: text('address'),
+    contractName: text('contract_name')
+  },
+  (chaindexingContractAddress) => ({
+    chaindexingContractAddressesChainAddressIndex: uniqueIndex(
+      'chaindexing_contract_addresses_chain_address_index'
+    ).on(chaindexingContractAddress.chainId, chaindexingContractAddress.address)
+  })
+);
```
Comment (Jurshsmith) [inline chaindexing-postgres/src/drizzle/schema.ts:28]: Nice 👍🏾

File: chaindexing-postgres/src/drizzle/schema.ts [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
-  address: text('address'),
-  contractName: text('contract_name')
-});
+export const chaindexingContractAddressesSchema = pgTable(
+  'chaindexing_contract_addresses',
+  {
+    id: serial('id').primaryKey(),
+    chainId: integer('chain_id'),
+    lastIngestedBlockNumber: integer('last_ingested_block_number'),
+    startBlockNumber: integer('start_block_number'),
```
Comment (Jurshsmith) [inline chaindexing-postgres/src/drizzle/schema.ts:19]: Sorry, forgot to mention. Some of these fields are stale.

```suggestion
    startBlockNumber: integer('start_block_number'),
    nextBlockNumberToIngestFrom: integer('next_block_number_to_ingest_from'),
    nextBlockNumberToHandleFrom: integer('next_block_number_to_handler_from'),
```

File: chaindexing-postgres/src/repo.postgres.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    await conn.insert(chaindexingContractAddressesSchema).values(values);
+  }
+
+  async streamContractAddresses(_conn: Conn, _streamer: any) {
+    // TODO: Implement
+    return Promise.resolve();
+  }
+
+  async createEvents(_conn: Conn, _events: any) {
```
Comment (Jurshsmith) [inline chaindexing-postgres/src/repo.postgres.ts:46]: Nit: `Event` interface exists

```suggestion
  async createEvents(_conn: Conn, _events: Event[]) {
```

File: chaindexing-repos/src/index.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ export abstract class Repo<Pool, Conn> {
   abstract getConn(): Promise<Conn>;
   abstract runInTransaction(
     conn: Conn,
-    repo_ops: (transaction_conn: Conn) => Promise<void>
+    repoOps: (transaction_conn: Conn) => Promise<void>
```
Comment (Jurshsmith) [inline chaindexing-repos/src/index.ts:12]: Nice 👍🏾

File: chaindexing-postgres/src/repo.postgres.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { Repo } from '@chaindexing/repos';
+import { NodePgDatabase, drizzle } from 'drizzle-orm/node-postgres';
+import { Pool } from 'pg';
+import { chaindexingContractAddressesSchema, pool } from './drizzle';
+
+type Conn = NodePgDatabase<Record<string, never>>;
+
+export class PostgresRepo extends Repo<Pool, Conn> {
+  pool: Pool;
+  conn?: Conn;
```
Comment (Jurshsmith) [inline chaindexing-postgres/src/repo.postgres.ts:11]: These should not exist in the Repo object. 

```suggestion
  
```

**Review by Jurshsmith — CHANGES_REQUESTED** [review]
Looks good. 🚀 Just a few tiny detail comments and updating the stale `ContractAddress` fields.

**Comment by conquext** [comment]
> Looks good. 🚀 Just a few tiny detail comments and updating the stale `ContractAddress` fields.

The issues have been resolved accordingly.

**Review by Jurshsmith — APPROVED** [review]

**Pull request merged**